gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/AbstractGhfsStatisticsSource.java

```diff
@@ -63,6 +63,16 @@ public long incrementCounter(String name, long value) {
     return ioStatistics.incrementCounter(name, value);
   }
 
+  /**
+   * set the value for a named counter.
+   *
+   * @param key counter name
+   * @param value value to set
+   */
+  public void setCounter(String key, long value) {
+    ioStatistics.setCounter(key, 0L);
+  }
```

Comment:
We should be setting `value` instead of 0

Reply:
Updated